internal/server/routes.go

```diff
@@ -282,9 +283,14 @@ func (a *API) notFoundHandler(c *gin.Context) {
 	}
 
 	c.Status(http.StatusNotFound)
-	buf, err := assetFS.ReadFile("ui/404.html")
-	if err != nil {
-		logging.S.Error(err)
+
+	buf := []byte("404 not found")
+	var err error
+	if a.server.options.UI.FS != nil {
+		buf, err = fs.ReadFile(a.server.options.UI.FS, "ui/static/404.html")
+		if err != nil {
+			logging.S.Error(err)
+		}
```

Comment:
This only works in tests, right? It will no longer serve the 404 page from the static files because we never set the FS on `server.options.UI.FS`.